### PR TITLE
Fix Pinot Timeboundary handling

### PR DESCRIPTION
**Pull request by elonazoulay**
## Description

Fix time boundary handling for server queries

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java [hunk 1]
```diff
@@ -128,6 +128,8 @@
     // Use a fixed instant for testing date time functions
     private static final Instant CREATED_AT_INSTANT = Instant.parse("2021-05-10T00:00:00.00Z");
 
+    private static final DateTimeFormatter MILLIS_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.of("UTC"));
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java:131]: Can we use `ZoneOffset.UTC` with static import?

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java [hunk 2]
```diff
@@ -445,69 +447,52 @@ private void createAndPopulateMixedCaseHybridTablesAndTopic(TestingKafka kafka,
...
             Files.createDirectory(temporaryDirectory);
             ImmutableList.Builder<GenericRow> offlineRowsBuilder = ImmutableList.builder();
-            for (int i = 8; i < 12; i++) {
+            for (int i = 4; i < 8; i++) {
                 GenericRow row = new GenericRow();
                 row.putValue("stringCol", "string_" + i);
-                row.putValue("longCol", (long) i - 8);
-                row.putValue("updatedAt", initialUpdatedAt.truncatedTo(DAYS).minusSeconds(3600 * 12 * (i - 8)).toEpochMilli());
+                row.putValue("longCol", (long) i);
+                row.putValue("updatedAt", startInstant.plusMillis(86_400_000 + 1000 * (i - 4)).toEpochMilli());
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java:482]: How about `startInstant.plus(1, DAYS).plusMillis(1000 * (i - 4)).toEpochMilli()`?  Same for other places.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java [hunk 3]
```diff
@@ -2514,4 +2499,20 @@ public void testHavingClause()
...
+
+    @Test
+    public void testTimeBoundary()
+    {
+        // Note: This table uses Pinot TIMESTAMP and not LONG as the time column type.
+        Instant startInstant = initialUpdatedAt.truncatedTo(DAYS);
+        String expectedValues = "(VARCHAR 'string_8', BIGINT '8', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 1000)) + "')," +
+                "(VARCHAR 'string_9', BIGINT '9', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 2000)) + "')," +
+                "(VARCHAR 'string_10', BIGINT '10', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 3000)) + "')," +
+                "(VARCHAR 'string_11', BIGINT '11', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 4000)) + "')";
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java:2509]: nit: We can add `VALUES` to `expectedValues`:
```java
        String expectedValues = "VALUES " +
                "(VARCHAR 'string_8', BIGINT '8', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 1000)) + "')," +
                "(VARCHAR 'string_9', BIGINT '9', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 2000)) + "')," +
                "(VARCHAR 'string_10', BIGINT '10', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 3000)) + "')," +
                "(VARCHAR 'string_11', BIGINT '11', TIMESTAMP '" + MILLIS_FORMATTER.format(startInstant.plusMillis(-86_400_000 + 4000)) + "')";
```

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java [hunk 4]
```diff
@@ -2514,4 +2499,20 @@ public void testHavingClause()
                 .matches("VALUES (VARCHAR 'Los Angeles', BIGINT '50000')")
                 .isFullyPushedDown();
     }
+
+    @Test
+    public void testTimeBoundary()
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java:2502]: nit: Move to before `testTimestamp()`.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**